kratos/containers/model.cpp

```diff
@@ -27,109 +27,229 @@
+// 
+//         }
+// 
+//         //add the root model part to the list
+//         ModelPart& root_model_part = pModelPart->GetRootModelPart();
+//         mRootModelPartMap[root_model_part.Name()] = &root_model_part;
+
+        KRATOS_CATCH("")
+    }
 
-        //add the root model part to the list
-        ModelPart& root_model_part = pModelPart->GetRootModelPart();
-        mRootModelPartMap[root_model_part.Name()] = &root_model_part;
+    void Model::AddModelPartRawPointer( ModelPart* pModelPart) //TODO: DEPRECATED. to be removed. this is TEMPORARY
+    {
+        KRATOS_TRY
+
+        KRATOS_WARNING("Model") << "Model::AddModelPartRawPointer is deprecated and is currently doing nothing" << std::endl;
+
+//         //TODO: flat map should disappear in the future!!
+//         auto search = mflat_map.find(pModelPart->Name());
+//         if( search == mflat_map.end())
+//         {ModelPartName
+//             mflat_map[pModelPart->Name()] = pModelPart;
+// 
+//             //walk the submodelparts
+//             for(auto& part : pModelPart->SubModelParts())
+//                 AddModelPartRawPointer(&part);
+//         }
+//         else
+//         {
+//             if(&(*search->second) != &*(pModelPart))
+//                 KRATOS_ERROR << "trying to add to the Model two DISTINCT model parts with the same name. This should be possible (and it will be in the future) if they belong to two different root model_parts, but it is currently disallowed";
+// 
+//         }
+// 
+//         //add the root model part to the list
+//         ModelPart& root_model_part = pModelPart->GetRootModelPart();
+//         mRootModelPartMap[root_model_part.Name()] = &root_model_part;
 
         KRATOS_CATCH("")
     }
 
     ModelPart& Model::GetModelPart(const std::string& rFullModelPartName)
     {
         KRATOS_TRY
+        
+//         KRATOS_INFO("Model") << "within GetModelPart address of Model is " <<  &(*this) << std::endl; //TODO: remove, this is for debugging purposes
 
         KRATOS_ERROR_IF( rFullModelPartName.empty() ) << "Attempting to find a "
             << "ModelPart with empty name (\"\")!" << std::endl;
-
-        auto search = mflat_map.find(rFullModelPartName);
-        if(search != mflat_map.end()) {
-            // TODO enable the warning
-            // KRATOS_WARNING_IF("Model", (search->second)->IsSubModelPart())
-            //     << "Getting a SubModelPart from the Model without "
-            //     << "specifying the RootModelPart is deprecated and will be removed\n"
-            //     << "Please use e.g \"RootModelPart.SubModelPart.SubSubModelPart\" "
-            //     << "as input for this function" << std::endl;
-            return *(search->second);
+            
+        std::vector< std::string > subparts_list;
+        GetSubPartsList(rFullModelPartName, subparts_list);
```

Comment:
Better to return as `std::vector< std::string > subparts_list = GetSubPartsList(rFullModelPartName)`.